lib/Model/Listing/Dao/QueryBuilderHelperTrait.php

```diff
@@ -23,13 +23,26 @@
 trait QueryBuilderHelperTrait
 {
     /**
-     * @var callable|null
+     * @var callable[]
      */
-    protected $onCreateQueryBuilderCallback;
+    protected array $queryBuilderProcessors = [];
 
     public function onCreateQueryBuilder(?callable $callback): void
     {
-        $this->onCreateQueryBuilderCallback = $callback;
+        $this->discardQueryBuilderProcessors();
+        if (is_callable($callback)) {
+            $this->addQueryBuilderProcessor($callback);
+        }
```

Comment:
Reverting 
`protected $onCreateQueryBuilderCallback;`
and
```suggestion
        $this->onCreateQueryBuilderCallback = $callback;
        $this->queryBuilderProcessors = [$callback];
```
```
        foreach ($this->queryBuilderProcessors as $processor) {
            if (is_callable($callback)) {
                $processor($queryBuilder);
            }
        }
```

should be fine, right?